Sources/eppo/AssignmentCache.swift

```diff
@@ -9,51 +9,56 @@ public struct AssignmentCacheKey {
     var subjectKey: String
     var flagKey: String
     var allocationKey: String
-    var variationValue: EppoValue
+    var variationKey: String
 }
 
 public class InMemoryAssignmentCache: AssignmentCache {
-    private var cache: [String: String] = [:]
-    
+    internal var cache: [CacheKey: CacheValue] = [:]
+
+    internal struct CacheKey: Hashable {
+        let subjectKey: String
+        let flagKey: String
+        
+        func hash(into hasher: inout Hasher) {
+            hasher.combine(subjectKey)
+            hasher.combine(flagKey)
+        }
+    }
+
+    internal struct CacheValue: Equatable {
+        let allocationKey: String
+        let variationKey: String
```

Comment:
Should we be hashing these two things as well? (This was the direction we're going in the JS SDK)

Reply:
Not necessary at this time as this is an in-memory cache; I have some follow-up work for storing to disk. In that case we will implement the `Decodable` interface here and an appropriate handler.